### ISSUE TITLE
graphite: web and api module do not work

### ISSUE THREAD

**Issue by makefu** [NixOS/nixpkgs#12467]
I wanted to use the graphite api and graphite web services for graphite but the services do not boot up in the latest unstable:

`$ journalctl -x -u graphiteApi`
```
waitress-serve[17821]: Error: Bad module 'graphite_api.app'
waitress-serve[17821]: Usage
...
```

and when trying to `services.graphite.web.enable`
`$ journalctl -x -u graphiteWeb`
```
waitress-serve[4540]: Error: Bad module 'django.core.handlers.wsgi'
waitress-serve[4540]: Usage
...
```

My graphite config:
```
  services.graphite = {
    api = {
      enable = true;
      listenAddress = "0.0.0.0";
    };
    carbon = {
      enableCache = true;
      # save disk usage by restricting to 1 bulk update per second
      config = ''
        [cache]
        MAX_CACHE_SIZE = inf
        MAX_UPDATES_PER_SECOND = 1
        MAX_CREATES_PER_MINUTE = 50
        '';
      storageSchemas = ''
        [carbon]
        pattern = ^carbon\.
        retentions = 60:90d

        [default]
        pattern = .*
        retentions = 60s:30d,300s:1y
        '';
    };
  };
```
I tried this module successfully around a year ago but i wanted to deploy it now.
Can anyone confirm this issue or am i holding the nixos config wrong?

/cc @offlinehacker

**Comment by makefu** [comment]
any updates on that one? it is actually pretty sad as i wanted to deploy graphite via nixos but now i have to rely on a legacy host (ubuntu) to provide the service.

If wanted i can try to create a nixos test case to avoid silent fails in the future.

**Comment by fpletz** [comment]
I have a graphite deployed with nixos which I want to update soon. I will fix it if no one beats me to it.

**Comment by makefu** [comment]
@fpletz any updates? sorry to be so pushy but i'd really love to deploy graphite with nixOS instead of still having to run it on my legacy host (archlinux)

**Comment by fpletz** [comment]
This weekend! :wink:

**Comment by makefu** [comment]
@fpletz Sorry to poke again, any updates on this one?

**Comment by makefu** [comment]
a quick follow-up. i found that graphiteApi and graphiteWeb themselve have dependencies which are note included when you simply set the PYTHONPATH. what i do is to use the same technique as in https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/misc/bepasty.nix#L107 (a module i maintain) to include all recursive dependencies.

i will submit a pull request for the changes shortly so do not bother with it.

Regards

**Comment by makefu** [comment]
Closed via #13854 .Thanks @globin for the merge

**Comment by basvandijk** [comment]
I'm also running into this bug on the `release-16.03` branch. 

@makefu, @globin can the fix be cherry picked into the 16.03 release?

**Comment by domenkozar** [comment]
done
